ls++

```diff
@@ -388,30 +412,44 @@ sub owner {
 
 sub size {
   my ($size) = @_;
+  if ($size =~ /\d+,\s*\d+/){ # not size but major/minor values for special files
+    $size = sprintf("%*s", $sizelen, $size);
+    return $size;
+  }
   #FIXME
   if($colors > 16) {
     #$size =~ s/(\S+)(K)/$c[2]$1\e[0m$c[4]$2\e[0m/gi;# and print "AA\n";
+    if($sizelen < 4){ # constraining size length between 4 and 8 characters max
+        $stringlen = 4;
+    }
+    elsif($sizelen >= 8){
+        $stringlen = 7;
+    }
+    else
+    {
+        $stringlen = $sizelen;
+    }
     if($size =~ m/^(\S+)(K)/) {
       $size = sprintf("%27s",
-        fg($c[7], sprintf("%4g", $1))
+        fg($c[7], sprintf("%*g", $stringlen, $1))
           . fg($c[2], fg('bold', $2))
         );
     }
     elsif($size =~ m/^(\S+)(M)/) {
       $size = sprintf("%29s",
-        fg($c[7], sprintf("%4g", $1))
+        fg($c[7], sprintf("%*g", $stringlen, $1))
           . fg($c[4], fg('bold', $2))
         );
     }
     elsif($size =~ m/^(\S+)(G)/) {
       $size = sprintf("%27s",
-        fg($c[7], sprintf("%4g", $1))
+        fg($c[7], sprintf("%*g", $stringlen, $1))
           . fg($c[3], fg('bold', $2))
         );
     }
     elsif($size =~ m/^(\d+)/) {
       $size = sprintf("%27s",
-        fg($c[7], sprintf("%4d", $1))
+        fg($c[7], sprintf("%*d", $stringlen, $1))
```

Comment:
$stringlen is finally used for more than one single thing now. :)
It avoids hardcoding "size" column width and dynamically changes it depending on the maximum size of the "size" string.